### DISCUSSION TITLE
Auto Maintain source of truth by having scheduled routine sync

### DISCUSSION THREAD

**Discussion by quiknick** [emersonfelipesp/netbox-proxbox#53] [Ideas]
I know Netbox team is against automated syncing, but in the real world where SecDevOps is in play, VMs are sometimes added on the fly, or during outages, or to support fast pace requirements that are sometimes outside the normal slow processes.  I would like routine automated syncs to Netbox that can be either triggered by a schedule time from this plugin.

**Comment by emersonfelipesp** [comment]
> I know Netbox team is against automated syncing, but in the real world where SecDevOps is in play, VMs are sometimes added on the fly, or during outages, or to support fast pace requirements that are sometimes outside the normal slow processes. I would like routine automated syncs to Netbox that can be either triggered by a schedule time from this plugin.

I think that the best solution for this problem is **Proxmox enable support for webhook**, in a way that it automatically sends updates to Proxbox. Instead of Proxbox pulling information from Proxmox, **Proxmox would send periodic updates to Proxbox**.

Although your idea is useful and a good feature, I really don't want to Sync it periodically by pulling Proxmox information. I will see if there's any effort to implement Webhook support on Proxmox and, if not, I will start a discussion there.

**Reply by quiknick** [reply]
I agree! A webhook when changes are made on Proxmox side and pushed it netbox is best. I originally had written it down, but thought maybe it was too much to ask. I did leave the either in the original request above. :+1:

**Reply by emersonfelipesp** [reply]
I have a lot of work to do related to syncing. When Proxbox be able to get manually absolutely all information available from Proxmox, then we come back here to see how we are going to solve it to automatically sync it. It would be great to have Webhook working until we get to this point.

**Comment by emersonfelipesp** [comment]
Hi @quiknick I [opened a Issue](https://github.com/netdevopsbr/netbox-proxbox/issues/112) about it and gonna closing this discussion.
